traffic_router/ultimate-test-harness/http_test.go

```diff
@@ -0,0 +1,497 @@
+	CoverageZoneLocation       string
+}
+
+var (
+	toConfig  TOConfig
+	toSession *client.Session
+	count     int
+)
+
+func getTOConfig(t *testing.T) {
+	err := envconfig.Process("", &toConfig)
+	if err != nil {
+		t.Fatalf("reading configuration from the environment: %s", err.Error())
+	}
+}
+
+var ipv4Only *bool
+var ipv6Only *bool
+var cdnName *string
+var deliveryServiceName *string
+var trafficRouterName *string
+var clientIPAddress *string
+var useCoverageZone *bool
+var coverageZoneLocation *string
+var requestsPerSecondThreshold *int
+var benchmarkTime *int
+var threadCount *int
+var pathCount *int
+var maxPathLength *int
+
+func init() {
+	rand.Seed(time.Now().UnixNano())
+	ipv4Only = flag.Bool("4", false, "test IPv4 addresses only")
+	ipv6Only = flag.Bool("6", false, "test IPv4 addresses only")
+	cdnName = flag.String("cdn", "", "the name of a CDN to search for Delivery Services")
+	deliveryServiceName = flag.String("ds", "", "the name (XMLID) of a Delivery Service to use for tests")
+	trafficRouterName = flag.String("hostname", "", "the hostname of a Traffic Router to use")
+	clientIPAddress = flag.String("ip_address", "", "spoof your client IP address to Traffic Router's geolocation")
+	useCoverageZone = flag.Bool("coverage_zone", false, "whether to use an IP address from the Traffic Router's Coverage Zone File")
+	coverageZoneLocation = flag.String("coverage_zone_location", "", "the Coverage Zone location to use (implies coverage_zone=true)")
+	requestsPerSecondThreshold = flag.Int("requests_threshold", 8000, "the minimum number of requests per second a Traffic Router must successfully respond to")
+	benchmarkTime = flag.Int("benchmark_time", 10, "the duration of each load test in seconds")
+	threadCount = flag.Int("thread_count", 12, "the number of threads to use for each test")
+	pathCount = flag.Int("path_count", 10000, "the number of paths to generate for use in requests to Delivery Services")
+	maxPathLength = flag.Int("max_path_length", 100, "the maximum length for each generated path")
+}
+
+func getCoverageZoneURL(cdnName tc.CDNName) (string, error) {
+	snapshot, _, err := toSession.GetCRConfig(string(cdnName), client.RequestOptions{})
+	if err != nil {
+		return "", fmt.Errorf("getting the Snapshot of CDN '%s': %s", cdnName, err.Error())
+	}
+	czPollingURLInterface, ok := snapshot.Response.Config[tc.CoverageZonePollingURL]
+	czPollingURL := czPollingURLInterface.(string)
+	if !ok {
+		return "", fmt.Errorf("parameters %s was not found in the Snapshot of CDN '%s'", tc.CoverageZonePollingURL, cdnName)
+	}
+	return czPollingURL, nil
+}
+
+func getCoverageZoneFile(czPollingURL string) (tc.CoverageZoneFile, error) {
+	czMap := tc.CoverageZoneFile{}
+	czMapRequest, err := http.NewRequest("GET", czPollingURL, nil)
+	if err != nil {
+		return czMap, fmt.Errorf("creating HTTP request for URL %s: %s", czPollingURL, err.Error())
+	}
+	czMapRequest.Header.Set("User-Agent", UserAgent)
+	httpClient := http.Client{}
```

Comment:
We should probably include a reasonable timeout for this request

Reply:
Set a timeout of 10 seconds in b52b52b146